geomstats/geometry/connection.py

```diff
@@ -188,34 +220,51 @@ def pole_ladder_step(self, base_point, next_point, base_shoot):
             base_point=next_point,
             tangent_vec=transported_tangent_vector)
 
-        return transported_tangent_vector, end_point
+        trajectories = []
+        if return_geodesics:
+            main_geodesic = self.geodesic(
+                initial_point=base_point,
+                end_point=next_point)
+            diagonal = self.geodesic(
+                initial_point=mid_point,
+                end_point=base_shoot)
+            final_geodesic = self.geodesic(
+                initial_point=next_point,
+                end_point=end_shoot)
+            trajectories = [main_geodesic, diagonal, final_geodesic]
+        return transported_tangent_vector, end_point, trajectories
```

Comment:
Do we want to return a blank list when return_geodesic is false? Or rather have a completely different return: `return transported_tangent_vector, end_point`?

Reply:
That's what I was wandering, the problem is that if you write that you have to adapt the way you then unpack the output l291 depending on the value of `return_geodesics` which will make the code longer and with repetitions...
I don't know what is the best practice here...